CIP-0159/README.md

```diff
@@ -0,0 +1,587 @@
+ADA and then the transaction gets processed at time `t1` when the account balance is 1000 ADA, the
+smart contracts in the transaction use different numbers in there executions. This can result in
+completely different execution budget usages which results in completely different fees. Same
+transaction; different fees.
+
+By using the account balance intervals instead, Alice's transaction would get the same inputs at
+`t0` and `t1`. As long as the intervals are still satisfied, it does not matter that the account
+balance has actually changed. Both executions result in the same execution budget usages and
+therefore, the same fees. Cardano's local determinism is preserved.
+
+### Whitelisted Native Assets Only
+
+The reason Cardano has a `minUTxOValue` requirement is to prevent native asset dusting attacks - a
+malicious person can create worthless tokens and create UTxOs with *only* these assets. When block
+producers try to validate transactions, they need to load the UTxOs into memory. If the size of
+these UTxOs are not somehow tied to real world resource constraints, the malicious actors can cause
+block producer memory usages to explode with these worthless native assets taking up space. This is
+effectively a ddos attack. By requiring every UTxO to contain ADA, a malicious actor is only able to
+cause this attack if they actually have the required amount of ADA to back the memory usage required
+to process the native asset values.
+
+If account addresses could accept arbitrary native assets like UTxOs, they would be susceptible to
+the same dusting attacks. A first thought might be to have a `minUTxOValue`-like protocol deposit
+for account addresses - the person depositing the native asset into the account would need to cover
+this protocol deposit. But this would actually prevent the account addresses from being used for
+some of the use cases discussed in the [Motivation section](#motivation-why-is-this-cip-necessary).
+Consider if a wallet was charging 0.1 ADA per transaction and decides to accept a stablecoin as
+payment instead. The protocol deposit could be more than 1 ADA which is 10x the amount the wallet
+wanted to charge! Users will not pay the fees in stablecoins if using ADA instead is 10x cheaper!
+
+So the goal is to prevent dusting attacks while still allowing micropayments using non-ADA assets.
+That is why the whitelist is used. The account address owner covers the protocol deposit for the native
+assets they are interested in and now users can just deposit those native assets without having to
+also cover a protocol deposit. The account address owner would not be able to cover the protocol
+deposit if they did not know which native assets would be deposited ahead of time, so this approach
+only works with a whitelist.
+
+## Path to Active
+
+### Phased Delivery
+
+In order to expedite delivery of this CIP, development will be broken over two phases/eras. *Each
+phase will require a new plutus and ledger version.*
```

Comment:
With suggestion I made earlier, there might not need to be a requirement for a new plutus version to support multi-assets. They likely could be enable retroactively for PlutusV4, once we implement the feature in the next Euler era (which will introduce PlutusV5, which will likely include other features that will not be available for PlutusV4).

That being said, this is up to the DApp developers if they want to proactively support multi-assets in accounts or not. So, if there is no desire for this can continue as planned and only add ADA to PlutusV4 context for deposits and balance intervals.

Reply:
yes please, unless Plutus V5 will land within a month of Plutus V4, taking steps to make it retroactive will be hugely helpful; we can start building and launch dApps that leverage this, that increase in usefulness when it gets turned on, without having to redeploy with recompiled scripts.

Reply:
( yea, that comment was mostly facetious ;) )